atlassian_jwt_auth/contrib/django/decorators.py

```diff
@@ -4,10 +4,54 @@
 from django.http.response import HttpResponse
 
 import atlassian_jwt_auth
-from .utils import parse_jwt, verify_issuers
+from .utils import parse_jwt, verify_issuers, _build_response
 from ..server.helpers import _requires_asap
 
 
+def validate_asap(issuers=None, subjects=None, required=True):
+    """Decorator to allow endpoint-specific ASAP authorization, assuming ASAP
+    authentication has already occurred.
+
+    :param list issuers: A list of issuers that are allowed to use the
```

Comment:
It feels odd to accept a list of `issuers` but then allow a function to be provided for the `subject`. My thoughts on this is what you extract out the `issuer` and the `subject` checks into the django asap classes to allow users to subclass and implement their own checks if that's desired.

Reply:
Issuers should be a relatively small list of other services, but subjects are whoever the service is acting for. If subjects were explicitly restricted to other services, then a list would make sense, but I can't find that in the spec.

A possible use case would be to have a service make a call on behalf of a user, where subject is set to an email address, a UUID, or something else that requires a function call to validate.

Reply:
I think I misunderstood your comment: pulling checks out into a class that could be overridden sounds fine, but I don't think there's currently a good place to do that.

Adding something like a `ClaimsValidator` class that could be used by the middleware and view decorators sounds fine to me, but out of scope for this PR.

So I can strip out the subjects function stuff for now, assuming a more comprehensive mechanism will be added later?

Reply:
I don't mind keeping it in. The problem I have is that it is adding complexity where we probably don't need it right now.

Reply:
Yeah, I'll just remove it.

Also: any opinions on the `ClaimsValidator` approach?

Reply:
Sounds okay to me but it might not be needed - re class methods.